tangelo/toolboxes/molecular_computation/integral_solver_pyscf.py

```diff
@@ -383,10 +392,18 @@ def compute_mean_field(self, sqmol):
             raise ValueError("Hartree-Fock calculation did not converge")
 
         if sqmol.symmetry:
-            sqmol.mo_symm_ids = list(self.symm.label_orb_symm(sqmol.mean_field.mol, sqmol.mean_field.mol.irrep_id,
+            if not sqmol.uhf:
+                sqmol.mo_symm_ids = list(self.symm.label_orb_symm(sqmol.mean_field.mol, sqmol.mean_field.mol.irrep_id,
                                                               sqmol.mean_field.mol.symm_orb, sqmol.mean_field.mo_coeff))
-            irrep_map = {i: s for s, i in zip(molecule.irrep_name, molecule.irrep_id)}
-            sqmol.mo_symm_labels = [irrep_map[i] for i in sqmol.mo_symm_ids]
+                irrep_map = {i: s for s, i in zip(molecule.irrep_name, molecule.irrep_id)}
+                sqmol.mo_symm_labels = [irrep_map[i] for i in sqmol.mo_symm_ids]
+            else:
+                for i in range(2):
```

Comment:
It is not clear at first glance why this loop is needed, and the use of `i` in your list and dict comprehensions below makes it confusing because it should prevail over it.

Are you sure you need a loop at all, considering the statements below do not use this `i` (I think) ?